java/compaction/compaction-job-creation/src/main/java/sleeper/compaction/job/creation/CreateCompactionJobs.java

```diff
@@ -121,31 +125,50 @@ public void createJobs(TableProperties table) throws StateStoreException, IOExce
     }
 
     private void createJobsForTable(TableProperties tableProperties, StateStore stateStore) throws StateStoreException, IOException, ObjectFactoryException {
+        Instant startTime = Instant.now();
         TableStatus table = tableProperties.getStatus();
         LOGGER.debug("Creating jobs for table {}", table);
 
+        Instant loadPartitionStartTime = Instant.now();
         List<Partition> allPartitions = stateStore.getAllPartitions();
+        Duration loadPartitionDuration = Duration.between(loadPartitionStartTime, Instant.now());
 
         // NB We retrieve the information about all the active file references and filter
         // that, rather than making separate calls to the state store for reasons
         // of efficiency and to ensure consistency.
+        Instant loadFilesStartTime = Instant.now();
         List<FileReference> fileReferences = stateStore.getFileReferences();
+        Duration loadFilesDuration = Duration.between(loadFilesStartTime, Instant.now());
 
+        Instant compactionStrategyCreationStartTime = Instant.now();
         CompactionStrategy compactionStrategy = objectFactory
                 .getObject(tableProperties.get(COMPACTION_STRATEGY_CLASS), CompactionStrategy.class);
+        Duration objectFactoryCreationDuration = Duration.between(compactionStrategyCreationStartTime, Instant.now());
         LOGGER.debug("Created compaction strategy of class {}", tableProperties.get(COMPACTION_STRATEGY_CLASS));
         CompactionJobFactory jobFactory = new CompactionJobFactory(instanceProperties, tableProperties, jobIdSupplier);
 
+        Instant indexCreationStartTime = Instant.now();
+        CompactionStrategyIndex index = new CompactionStrategyIndex(tableProperties.getStatus(), fileReferences, allPartitions);
+        Duration indexCreationDuration = Duration.between(indexCreationStartTime, Instant.now());
+
+        Instant jobCreationStartTime = Instant.now();
```

Comment:
Is it necessary to get the current time twice like this? It might be worth just holding the Instants rather than the Durations?